index.js

```diff
@@ -1,102 +1,53 @@
-
-    // Disable features with a minSupported version.
-    // This key is not supported in v1 and any features with this key
-    // will need to be disabled.
-    for (const feature of Object.keys(v1Config.features)) {
-        if (v1Config.features[feature].minSupportedVersion) {
-            delete v1Config.features[feature].minSupportedVersion
-            v1Config.features[feature].state = 'disabled'
-        }
-    }
-
-    return v1Config
-}
-
 const unprotectedListName = 'unprotected-temporary.json'
 
 // Grab all exception lists
-const jsonListNames = fs.readdirSync(LISTS_DIR).filter(listName => {
+const jsonListNames = fs.readdirSync(constants.LISTS_DIR).filter(listName => {
     return listName !== unprotectedListName && listName !== '_template.json'
 })
 for (const jsonList of jsonListNames) {
-    const listData = JSON.parse(fs.readFileSync(`${LISTS_DIR}/${jsonList}`))
+    const listData = JSON.parse(fs.readFileSync(`${constants.LISTS_DIR}/${jsonList}`))
     const configKey = jsonList.replace(/[.]json$/, '').replace(/-([a-z0-9])/g, function (g) { return g[1].toUpperCase() })
 
     delete listData._meta
     defaultConfig.features[configKey] = listData
 }
 
-const unprotectedDomains = new Set()
-function addExceptionsToUnprotected (exceptions) {
-    for (const exception of exceptions) {
-        unprotectedDomains.add(exception.domain)
-    }
-    return exceptions.map((obj) => obj.domain)
-}
-
-const listData = JSON.parse(fs.readFileSync(`${LISTS_DIR}/${unprotectedListName}`))
+// Setup initial unprotected list
+const listData = JSON.parse(fs.readFileSync(`${constants.LISTS_DIR}/${unprotectedListName}`))
 defaultConfig.unprotectedTemporary = listData.exceptions
 
-addExceptionsToUnprotected(defaultConfig.unprotectedTemporary)
-addExceptionsToUnprotected(defaultConfig.features.contentBlocking.exceptions)
+// Add initial exceptions to legacy files
+legacyFilesGenerator.addExceptionsToUnprotected(defaultConfig.unprotectedTemporary)
+legacyFilesGenerator.addExceptionsToUnprotected(defaultConfig.features.contentBlocking.exceptions)
 
 // Create generated directory
-mkdirIfNeeded(GENERATED_DIR)
+utils.mkdirIfNeeded(constants.GENERATED_DIR)
 // Create version directories
-mkdirIfNeeded(`${GENERATED_DIR}/v1`)
-mkdirIfNeeded(`${GENERATED_DIR}/v2`)
-
-function isFeatureMissingState (feature) {
-    return !('state' in feature)
-}
+utils.mkdirIfNeeded(`${constants.GENERATED_DIR}/v1`)
+utils.mkdirIfNeeded(`${constants.GENERATED_DIR}/v2`)
 
 const platformConfigs = {}
 
 // Handle platform specific overrides and write configs to disk
 for (const platform of platforms) {
     const platformConfig = JSON.parse(JSON.stringify(defaultConfig))
-    const overridePath = `${OVERRIDE_DIR}/${platform}-override.json`
+    const overridePath = `${constants.OVERRIDE_DIR}/${platform}-override.json`
 
     if (!fs.existsSync(overridePath)) {
-        writeConfigToDisk(platform, platformConfig)
+        utils.writeConfigToDisk(platform, platformConfig)
```

Comment:
This seems like a *existing* bug, we're not passing the second argument and we're creating the generated config files if the override doesn't exist?!
Probably can be deleted.